jina/peapods/runtimes/container/__init__.py

```diff
@@ -263,3 +280,135 @@ def _is_container_alive(self) -> bool:
         except docker.errors.NotFound:
             return False
         return True
+
+    # Static methods used by the Pea to communicate with the `Runtime` in the separate process
+    @staticmethod
+    def wait_ready_or_shutdown(
+        timeout: Optional[float],
+        ready_or_shutdown_event: Union['multiprocessing.Event', 'threading.Event'],
+        **kwargs,
+    ):
+        """
+        Check if the runtime has successfully started
+
+        :param timeout: The time to wait before readiness or failure is determined
+        :param ready_or_shutdown_event: the multiprocessing event to detect if the process failed or succeeded
+        :param kwargs: extra keyword arguments
+
+        :return: True if is ready or it needs to be shutdown
+        """
+        return ready_or_shutdown_event.wait(timeout)
```

Comment:
Seems like this is the default method implementation, which is only changed by `Zedlet`. Can we put this implementation in the base class and only let the `Zedlet` overwrite it? Much less verbose code. Same might hold true for the `_wait_for_cancel` method, even thought I am not utterly sure.

Reply:
I am not so sure how inherited methods behave with static method. I agree there is place for improvement, but can be done in a next PR